### PR TITLE
dbeaver/dbeaver#36036-sqlite-fix-column-without-datatye

### LINKED ISSUE #36036: SQLite: impossible to drop a column when a column without data type exists
### Description

SQLite: impossible to drop a column when a column without data type exists. E.g.

`CREATE TABLE testnull (
	id INTEGER,
	"null" null,
	"notnull" INTEGER
);`

### DBeaver Version

DBeaver 24.2.4 ea

### Operating System

Win11

### Database and driver

SQLite

### Steps to reproduce

1. Create a table with 3 columns, one of the columns doesn't have a data type
2. Delete a column except the column without a data type
ER: the column is deleted
AR: the deletion failed. The reason is in wrong syntax (n/a in the script):

```
CREATE TEMPORARY TABLE temp AS
SELECT
  Column1,
  Column3
FROM test;

DROP TABLE test;

CREATE TABLE test (
	Column1 INTEGER NOT NULL,
	Column3 **N/A**
);

INSERT INTO test
 (Column1,
  Column3)
SELECT
  Column1,
  Column3
FROM temp;

DROP TABLE temp;
```

### Additional context

https://github.com/user-attachments/assets/e7f8d7ce-1a1e-4e20-95c7-86062d465acf

### PR DESCRIPTION
**Pull request by eusebe-cda**
adding the ANY type (as in the documentation concerning STRICT tables: https://www.sqlite.org/stricttables.html) and apply it for existing columns without a known type

### CODE REVIEW

File: plugins/org.jkiss.dbeaver.ext.sqlite/src/org/jkiss/dbeaver/ext/sqlite/model/SQLiteMetaModel.java [hunk 1]
```diff
@@ -217,6 +217,8 @@ public boolean handleSequenceCacheReadingError(Exception error) {
 
     @Override
     public GenericTableColumn createTableColumnImpl(@NotNull DBRProgressMonitor monitor, JDBCResultSet dbResult, @NotNull GenericTableBase table, String columnName, String typeName, int valueType, int sourceType, int ordinalPos, long columnSize, long charLength, Integer scale, Integer precision, int radix, boolean notNull, String remarks, String defaultValue, boolean autoIncrement, boolean autoGenerated) {
+        typeName = (typeName == "N/A") ? "ANY" : typeName; //$NON-NLS-1$
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.ext.sqlite/src/org/jkiss/dbeaver/ext/sqlite/model/SQLiteMetaModel.java]: Did you test if this even works? In Java, you can't compare strings using `==`. You must use `.equals`.

Comment (ShadelessFox) [reply plugins/org.jkiss.dbeaver.ext.sqlite/src/org/jkiss/dbeaver/ext/sqlite/model/SQLiteMetaModel.java]: By the way, where does that `N/A` come from? I'd fix it there, not here.

**Review by ShadelessFox — CHANGES_REQUESTED** [review]
Thanks for the contribution. Please resolve the issues mentioned in the comments.

File: plugins/org.jkiss.dbeaver.ext.sqlite/src/org/jkiss/dbeaver/ext/sqlite/model/SQLiteMetaModel.java (hunk 1 above)
Comment (eusebe-cda) [reply plugins/org.jkiss.dbeaver.ext.sqlite/src/org/jkiss/dbeaver/ext/sqlite/model/SQLiteMetaModel.java]: yes, it works... I have seen the codacy issues and correct them this evening.

The "N/A" come from the GenericMetaModel class :
https://github.com/dbeaver/dbeaver/blob/363c07ab6b023e73bbb53fd91039c44526e89161/plugins/org.jkiss.dbeaver.ext.generic/src/org/jkiss/dbeaver/ext/generic/model/meta/GenericMetaModel.java#L773C1-L785C10

You're right, I can create a getDefaultTypeName() method in the GenericMetaModel, overrided in the SQLiteMetaModel class...

Comment (eusebe-cda) [reply plugins/org.jkiss.dbeaver.ext.sqlite/src/org/jkiss/dbeaver/ext/sqlite/model/SQLiteMetaModel.java]: > You're right, I can create a getDefaultTypeName() method in the GenericMetaModel, overrided in the SQLiteMetaModel class...

I made these changes

File: plugins/org.jkiss.dbeaver.ext.generic/src/org/jkiss/dbeaver/ext/generic/model/meta/GenericMetaModel.java [hunk 2]
```diff
@@ -1061,4 +1061,7 @@ public boolean supportsViews(@NotNull GenericDataSource dataSource) {
             ((JDBCDataSourceInfo) dataSourceInfo).supportsViews();
     }
 
+	protected String getDefaultTypeName() {
+		return "N/A";  //$NON-NLS-1$
+	}
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.ext.generic/src/org/jkiss/dbeaver/ext/generic/model/meta/GenericMetaModel.java]: Please use spaces and specify the `@NotNull` annotation for the return type.
```suggestion
    @NotNull
    protected String getDefaultTypeName() {
        return "N/A";  //$NON-NLS-1$
    }
```

**Review by ShadelessFox — APPROVED** [review]

**Review by E1izabeth — APPROVED** [review]

File: plugins/org.jkiss.dbeaver.ext.generic/src/org/jkiss/dbeaver/ext/generic/model/meta/GenericMetaModel.java (hunk 2 above)
Comment (eusebe-cda) [reply plugins/org.jkiss.dbeaver.ext.generic/src/org/jkiss/dbeaver/ext/generic/model/meta/GenericMetaModel.java]: Thanks for your review, and sorry for my dirty code...

**Comment by ShadelessFox** [comment]
Thanks for the contribution! Passing the PR to our QA team now.

**Pull request merged**